api/public/v2/test_results/views.py

```diff
@@ -0,0 +1,106 @@
+import django_filters
+from django_filters.rest_framework import DjangoFilterBackend
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework import mixins, status, viewsets
+from rest_framework.response import Response
+
+from api.shared.mixins import RepoPropertyMixin
+from api.shared.permissions import RepositoryArtifactPermissions
+from reports.models import TestInstance
+
+from .serializers import TestInstanceSerializer
+
+
+class TestResultsFilters(django_filters.FilterSet):
+    commit_id = django_filters.CharFilter(field_name="commitid")
+    outcome = django_filters.CharFilter(field_name="outcome")
+    duration_min = django_filters.NumberFilter(
+        field_name="duration_seconds", lookup_expr="gte"
+    )
+    duration_max = django_filters.NumberFilter(
+        field_name="duration_seconds", lookup_expr="lte"
+    )
+    branch = django_filters.CharFilter(field_name="branch")
+
+    class Meta:
+        model = TestInstance
+        fields = ["commit_id", "outcome", "duration_min", "duration_max", "branch"]
+
+
+@extend_schema(
+    parameters=[
+        OpenApiParameter(
+            "commit_id",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description="Commit SHA for which to return test results",
+        ),
+        OpenApiParameter(
+            "outcome",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description="Status of the test (failure, skip, error, pass)",
+        ),
+        OpenApiParameter(
+            "duration_min",
+            OpenApiTypes.INT,
+            OpenApiParameter.QUERY,
+            description="Minimum duration of the test in seconds",
+        ),
+        OpenApiParameter(
+            "duration_max",
+            OpenApiTypes.INT,
+            OpenApiParameter.QUERY,
+            description="Maximum duration of the test in seconds",
+        ),
+        OpenApiParameter(
+            "branch",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description="Branch name for which to return test results",
+        ),
+    ],
+    tags=["Test Results"],
+    summary="Retrieve test results",
+)
+class TestResultsView(
+    viewsets.GenericViewSet,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    RepoPropertyMixin,
+):
+    serializer_class = TestInstanceSerializer
+    permission_classes = [RepositoryArtifactPermissions]
+    filter_backends = [DjangoFilterBackend]
+    filterset_class = TestResultsFilters
+
+    def get_queryset(self):
```

Comment:
Would it be right if the entire function just be?
```
def get_queryset(self):
    return TestInstance.objects.filter(repoid=self.repo.repoid)
```
It should be the case that a Repository instance would always have a repoid.

Also `return TestInstance.objects.none()` would this return `None` when the callers would always expect an iterable?

Reply:
makes sense!